### PR TITLE
HMRC-Trade

### PR DESCRIPTION
**Pull request by santhosh-thangavel**
1.Rewritten pipeline using Databaker
2.Added cubes class
3.Added tracer

### CODE REVIEW

File: datasets/HMRC_Trade/main.py [hunk 1]
```diff
@@ -6,338 +6,132 @@
...
 
-yr = scraper.title.index("20")
-yr = scraper.title[yr:yr+4]
-yr
+scraper.select_dataset(title = lambda x: x.endswith('data tables'), latest = True)
+scraper
+scraper.dataset.family = info["families"]
 
-myfile = scraper.distributions[0].as_pandas(sheet_name=None)
+dist = scraper.distributions[1].downloadURL
```
Comment (Shannon95) [inline datasets/HMRC_Trade/main.py:35]: change selecting distirbution by title name and not index. e.g. :
`dist = scraper.distribution(title = lambda t : 'data tables' in t).downloadURL`

File: datasets/HMRC_Trade/main.py [hunk 2]
```diff
@@ -6,338 +6,132 @@
...
-# +
-sheets = ['1. Industry Group', '2. Age Group','3. Business Size']
-
-grops = 'HMRC Industry'
-types = 'HMRC Trade Statistic Type'
-flows = 'Flow'
+for name, tab in tabs.items():
+    if 'Notes and Contents' in name or '5. Metadata' in name :
+        continue
+    datasetTitle = "uk-trade-in-goods-by-business-characteristics-2019"
```
Comment (Shannon95) [inline datasets/HMRC_Trade/main.py:44]: use distribution.title to retrieve title, helps to ensure no human error.
e.g. :
datasetTitle = distribution.title

File: datasets/HMRC_Trade/main.py [hunk 3]
```diff
@@ -6,338 +6,132 @@
...
+    if len(date) == 5:
+        return 'year/' + left(date, 4)
+df['Year'] = df['Year'].astype(str).replace('\.', '', regex=True)
+df['Year'] = df['Year'].apply(date_time)
+trace.Year("Formating to be year/2019")
+
+with pd.option_context('float_format', '{:f}'.format):
+    print(df)
+
+cubes.add_cube(scraper, df.drop_duplicates(), "uk-trade-in-goods-by-business-characteristics-2019")
```
Comment (Shannon95) [inline datasets/HMRC_Trade/main.py:134]: change to pass the distribution title also e.g :
cubes.add_cube(scraper, tidy.drop_duplicates(), distribution.title )

**Review by Shannon95 — COMMENTED** [review]
All looks fine except for a few small changes on defining distribution and title. Once changed happy to be merged

File: datasets/HMRC_Trade/main.py [hunk 4]
```diff
@@ -6,338 +6,132 @@
...
+trace = TransformTrace()
+cubes = Cubes('info.json')
 # -
 
-yr = scraper.title.index("20")
-yr = scraper.title[yr:yr+4]
-yr
+scraper.select_dataset(title = lambda x: x.endswith('data tables'), latest = True)
+scraper
+scraper.dataset.family = info["families"]
```
[the remark this answers is not in the document]
Comment (santhosh-thangavel) [reply datasets/HMRC_Trade/main.py:33]: Kept the line as Scraper(seed = 'info.json) is not picking up trade family.

File: datasets/HMRC_Trade/main.py (hunk 1 above)
Comment (santhosh-thangavel) [reply datasets/HMRC_Trade/main.py]: Made changes to get the latest distribution

File: datasets/HMRC_Trade/main.py (hunk 2 above)
Comment (santhosh-thangavel) [reply datasets/HMRC_Trade/main.py]: Title is retrieved through API

File: datasets/HMRC_Trade/main.py (hunk 3 above)
Comment (santhosh-thangavel) [reply datasets/HMRC_Trade/main.py]: distribution title is passed to cubes.add_cube() function

**Pull request merged**